### PR TITLE
feat: make `Log` SDK v3 compatible #major

### PR DESCRIPTION
**Pull request by RobinBol**
Note this is breaking w.r.t. to SDK v2.

### CODE REVIEW

File: lib/Log.js [hunk 1]
```diff
@@ -1,155 +1,231 @@
...
-		if (!err && typeof result === 'string') {
-			this.setTags({ homeyId: result })
-		}
-	}
-
-	setTags( tags ) {
-		if( this._client ) {
-			this._client.setTagsContext(tags);
-		}
+const Raven = require('raven');
```
Comment (tjallingt) [inline lib/Log.js:5]: not relevant for this PR but should we upgrade to `@sentry/node` ?

Comment (RobinBol) [reply lib/Log.js:5]: Yes, good question, that was also suggested here https://github.com/athombv/node-homey-log/pull/6, however, it appears to have an install size of ~4.2MB which I think is too much for a Homey App. Maybe if they manage to get the install size down to a reasonable number we can consider to upgrade.

Comment (tjallingt) [reply lib/Log.js:5]: Yea its install size is HUGE but its actual bundle size is apparently not that big...
 https://packagephobia.com/result?p=%40sentry%2Fnode
Seems like it is shipping a bunch of extra files and their monorepo structure doesn't help in this case :/

So we could use the latest and greatest but only if we use a bundler

File: package.json [hunk 2]
```diff
@@ -1,26 +1,51 @@
...
+    "lint": "eslint .",
+    "serve": "concurrently \"serve build/docs/\" \"npm run jsdoc:watch\"",
+    "build": "npm run jsdoc",
+    "jsdoc": "cd docs; jsdoc -c jsdoc.json",
+    "jsdoc:watch": "npm-watch jsdoc"
   },
   "repository": {
     "type": "git",
-    "url": "git+https://WeeJewel@github.com/athombv/node-homey-log.git"
+    "url": "git+https://https://github.com/athombv/node-homey-log.git"
```
Comment (aivus) [inline package.json:16]: Double `https://`

Comment (RobinBol) [reply package.json:16]: Thanks, fixed

**Pull request merged**